File: exams/frmNationalExaminationsEntryPrompt.vb
Public Class frmNationalExaminationsEntryPrompt


    Private Sub btnEnter_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnEnter.Click
        If cboYear.SelectedItem.ToString <> None Then
            If cboExamination.SelectedItem <> None Then
                exam_name = cboExamination.SelectedItem
                cont = True
                yr = cboYear.SelectedItem
                load_from_alumni = CheckBox1.Checked
                Me.Close()
            Else
                failure("Invalid Choice For Examination Name!")
            End If
        Else
            failure("Invalid Choice For Year!")
        End If
    End Sub

    Private Sub cboYear_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cboYear.SelectedIndexChanged
        If cboYear.SelectedItem.ToString <> None Then
            If stud Then
                query = "SELECT Name FROM national_examinations WHERE Year='" & cboYear.SelectedItem & "'"
            Else
                query = "SELECT Name FROM national_examinations WHERE Year='" & cboYear.SelectedItem & "'"
            End If
            If qread(query) Then
                While dbreader.Read
                    cboExamination.Items.Add(dbreader("Name"))
                End While
            End If
        End If
    End Sub

    Private Sub frmNationalExaminationsEntryPrompt_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            For k As Integer = startyear To endyear
                cboYear.Items.Add(k)
            Next
            cboYear.SelectedItem = Today.Year
        End If
    End Sub
End Class

File: exams/frmNationalExaminationsEntryPrompt.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmNationalExaminationsEntryPrompt
    {
        public frmNationalExaminationsEntryPrompt()
        {
            InitializeComponent();
            _cboYear.Name = "cboYear";
            _btnEnter.Name = "btnEnter";
        }

        private void btnEnter_Click(object sender, EventArgs e)
        {
            if ((cboYear.SelectedItem.ToString() ?? "") != (publicSubsNFunctions.None ?? ""))
            {
                if (Conversions.ToBoolean(Operators.ConditionalCompareObjectNotEqual(cboExamination.SelectedItem, publicSubsNFunctions.None, false)))
                {
                    publicSubsNFunctions.exam_name = Conversions.ToString(cboExamination.SelectedItem);
                    publicSubsNFunctions.cont = true;
                    publicSubsNFunctions.yr = Conversions.ToInteger(cboYear.SelectedItem);
                    publicSubsNFunctions.load_from_alumni = CheckBox1.Checked;
                    Close();
                }
                else
                {
                    publicSubsNFunctions.failure("Invalid Choice For Examination Name!");
                }
            }
            else
            {
                publicSubsNFunctions.failure("Invalid Choice For Year!");
            }
        }

        private void cboYear_SelectedIndexChanged(object sender, EventArgs e)
        {
            if ((cboYear.SelectedItem.ToString() ?? "") != (publicSubsNFunctions.None ?? ""))
            {
                if (publicSubsNFunctions.stud)
                {
                    publicSubsNFunctions.query = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("SELECT Name FROM national_examinations WHERE Year='", cboYear.SelectedItem), "'"));
                }
                else
                {
                    publicSubsNFunctions.query = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("SELECT Name FROM national_examinations WHERE Year='", cboYear.SelectedItem), "'"));
                }

                if (publicSubsNFunctions.qread(ref publicSubsNFunctions.query))
                {
                    while (publicSubsNFunctions.dbreader.Read())
                        cboExamination.Items.Add(publicSubsNFunctions.dbreader["Name"]);
                }
            }
        }

        private void frmNationalExaminationsEntryPrompt_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                for (int k = publicSubsNFunctions.startyear, loopTo = publicSubsNFunctions.endyear; k <= loopTo; k++)
                    cboYear.Items.Add(k);
                cboYear.SelectedItem = DateAndTime.Today.Year;
            }
        }
    }
}